### PR TITLE
[MVS-498] connect a QR code generator to a text field

### PR DESCRIPTION
**Pull request by ShahdSawalhi**
## :floppy_disk: [Code Check]
_Make sure you've checked/done the following before making this PR!_
* Does your code build/run?
* Does your design approach make sense?
* Does your work achieve the intention of the card?
* Does your code have any unintentional side effects -- have you tested a full workflow & checked the logic branches you touched?
* Did you make a unit test for your code (as appropriate)?
* Is your branch relatively up to date with the Development branch (within ~2 commits)
* Have you moved the card to "Needs QA" in Jira

## :flipper: [JIRA Sprint Card]

https://mass-immunization-system.atlassian.net/browse/MVS-498

## :newspaper: [Work Description]

Generate a QR code depending on the input of a text field. When you write inside the text field, you will get a QR code. Once the QR code is scanned it will give you the text field's input. You will also have the option to open the QR code on a separate tab.

## :vertical_traffic_light: [Testing/QA Notes]

- Run the Patient Registration app locally
- Click submit and go to the Single Patient Follow up page 
- Type something in the text field; a QR code should be generated once you do so 
- scan the QR code and make sure it gives the same input as the text field

### CODE REVIEW

**Comment by TomMakes** [comment]
Ran and found it fully functional!
Tested on Firefox browser.
Tested small and large (700 character) QR codes, both worked.
Didn't see any errors in the console log relating to the QR generation.

File: PatientRegistrationApp/src/components/SinglePatientFollowUp.vue [hunk 1]
```diff
@@ -1,23 +1,92 @@
...
+    Open 
+    </v-btn>
+
     </v-row>
 	</v-container>
 </template>
 
 <script>
-
+import qrCode from 'qrcode'
```
Comment (david-fraley) [inline PatientRegistrationApp/src/components/SinglePatientFollowUp.vue:42]: Is this where we typically leave import statements?

File: PatientRegistrationApp/yarn.lock [hunk 2]
```diff
@@ -2564,7 +2569,25 @@ bser@2.1.1:
   dependencies:
     node-int64 "^0.4.0"
 
-buffer-from@1.x, buffer-from@^1.0.0:
+buffer-alloc-unsafe@^1.1.0:
+  version "1.1.0"
+  resolved "https://registry.yarnpkg.com/buffer-alloc-unsafe/-/buffer-alloc-unsafe-1.1.0.tgz#bd7dc26ae2972d0eda253be061dba992349c19f0"
```
Comment (david-fraley) [inline PatientRegistrationApp/yarn.lock:2574]: What are all these changes to the yarn.lock file?

File: PatientRegistrationApp/src/components/SinglePatientFollowUp.vue [hunk 3]
```diff
@@ -1,23 +1,92 @@
...
+    </v-row>
+
     <v-row>
-QR Code Page
+    <v-btn
+      color="accent"
+      :disabled="!qrSrc"
+      @click="reset"
+    >
+    Reset
```
Comment (david-fraley) [inline PatientRegistrationApp/src/components/SinglePatientFollowUp.vue:26]: We should anchor the "reset" and "open" buttons to the bottom of the page to be consistent with our UI. I'd be ok not doing that if these buttons will change in the future. 
image

**Review by david-fraley — CHANGES_REQUESTED** [review]
You should update the PR itself so the title does have a typo and notify people they'll need to install another package in order to run PatReg once this gets merged in.

File: PatientRegistrationApp/src/components/SinglePatientFollowUp.vue (hunk 3 above)
Comment (ShahdSawalhi) [reply PatientRegistrationApp/src/components/SinglePatientFollowUp.vue:26]: These buttons will be changed later to Download & Send, once the QR code generator is connected to the ID generator.

File: PatientRegistrationApp/yarn.lock (hunk 2 above)
Comment (ShahdSawalhi) [reply PatientRegistrationApp/yarn.lock:2574]: Got that change when I added the qrcode library

File: PatientRegistrationApp/src/components/SinglePatientFollowUp.vue (hunk 1 above)
Comment (ShahdSawalhi) [reply PatientRegistrationApp/src/components/SinglePatientFollowUp.vue:42]: yes, that's what we do for the event bus. We import it in every page it is used in. Therefore, you need to import it in whatever page the qrcode is used in

**Review by david-fraley — CHANGES_REQUESTED** [review]
Good to go thanks

**Review by david-fraley — APPROVED** [review]

**Pull request merged**